fontbe/src/features.rs

```diff
@@ -553,23 +553,20 @@ impl Work<Context, AnyWorkId, Error> for FeatureCompilationWork {
         let marks = context.fea_rs_marks.get();
 
         let mut result = self.compile(&static_metadata, &ast, kerns.as_ref(), marks.as_ref())?;
-        if result.gdef_classes.is_none() {
+        if result.gdef_classes.is_none() && !static_metadata.gdef_categories.categories.is_empty() {
             // there were no explicit gdef categories, so let's use the ones
             // that were in public.openTypeCatgories or which we computed from
             // GlyphData.xml
```

Comment:
it's slightly difficult to line up the comment with the not-quite-matching fields. Can you update to specify. For example, is static_metadata.gdef_categories not the explicit gdef_categories? Is it the ones from public.openTypeCategories?

Reply:
updated this comment to better explain what's going on.